soccer/src/soccer/strategy/agent/position/seeker.cpp

```diff
@@ -0,0 +1,152 @@
+#include "seeker.hpp"
+
+namespace strategy {
+
+Seeker::Seeker(int seeker_num) {
+    offense_type_ = "Seeker";
+    seeker_pos_ = seeker_num;
+}
+
+std::optional<RobotIntent> Seeker::get_task(RobotIntent intent, const WorldState* world_state,
+                                            FieldDimensions field_dimensions) {
+    rj_geometry::Point current_loc = world_state->our_robots[intent.robot_id].pose.position();
+    rj_geometry::Point target_pt = get_open_point(world_state, current_loc, field_dimensions);
+
+    // stop at end of path
+    rj_geometry::Point target_vel{0.0, 0.0};
+
+    planning::PathTargetFaceOption face_option = planning::FaceBall{};
+
+    bool ignore_ball = false;
+
+    planning::LinearMotionInstant goal{target_pt, target_vel};
+    intent.motion_command = planning::MotionCommand{"path_target", goal, face_option, ignore_ball};
+    return intent;
+}
+
+/*
+ * Gets a point nearby to the robot that the opposing robots have minimal line-of-sight on
+ */
+rj_geometry::Point Seeker::get_open_point(const WorldState* world_state,
+                                          rj_geometry::Point current_loc,
+                                          FieldDimensions field_dimensions) {
+    return Seeker::calculate_open_point(1.0, .2, current_loc, world_state, field_dimensions);
+}
+
+/*
+ * Iteratively calculates an optimal point with minimal line of sight by focusing on smaller and
+ * smaller areas
+ */
+rj_geometry::Point Seeker::calculate_open_point(double current_prec, double min_prec,
+                                                rj_geometry::Point current_point,
+                                                const WorldState* world_state,
+                                                FieldDimensions field_dimensions) {
+    if (current_prec < min_prec) {
+        return current_point;
+    }
+    rj_geometry::Point ball_pos = world_state->ball.position;
+    rj_geometry::Point min = current_point;
+    double min_val = max_los(ball_pos, current_point, world_state);
+    double curr_val;
```

Comment:
```suggestion
    double curr_val{};
```
is safer to initialize variables, as C++ has unexpected behavior if this is ever uninitialized